Reliability/300_Testing_for_Resiliency_of_EC2_RDS_and_S3/Code/Python/WebAppLambda/deploy_web_lambda.py

```diff
@@ -72,30 +80,56 @@ def process_global_vars():
         stackname = 'WebServersforResiliencyTesting'
     except SystemExit:
         sys.exit(1)
-    except:
+    except Exception:
         logger.error("Unexpected error!\n Stack Trace:", traceback.format_exc())
 
+
+def find_latest_ami_name(region, arch):
+    assert region, "Region is not defined"
+    assert arch, "Architecture is not defined"
+    assert arch in ARCH_TO_AMI_NAME_PATTERN, \
+        "Architecture must be one of {}".format(
+            ARCH_TO_AMI_NAME_PATTERN.keys())
+    pattern, owner = ARCH_TO_AMI_NAME_PATTERN[arch]
+    ec2 = boto3.client("ec2", region_name=region)
+    images = ec2.describe_images(
```

Comment:
This will not work unless you  add ec2:DescribeImages permission to the policy used by the DeployResiliencyWorkshop-WebAppLambdaRole-xxx 

---
```
  [INFO]	2019-09-20T13:08:35.415Z	69354144-9dac-4500-ad66-a0dcd4dc9169	{"log_level": "DEBUG", "region_name": "us-east-2", "secondary_region_name": "us-west-2", "cfn_region": "us-east-2", "cfn_bucket": "aws-well-architected-labs-ohio", "folder": "Reliability/", "workshop": "300-ResiliencyofEC2RDSandS3", "boot_bucket": "aws-well-architected-labs-ohio", "boot_prefix": "Reliability/", "boot_object": "bootstrap300Resiliency.sh", "websiteimage": "https://s3.us-east-2.amazonaws.com/arc327-well-architected-for-reliability/Cirque_of_the_Towers.jpg", "vpc": {"stackname": "ResiliencyVPC", "status": "CREATE_COMPLETE"}, "rds": {"stackname": "MySQLforResiliencyTesting", "status": "CREATE_COMPLETE"}}
…
File "/var/task/deploy_web_lambda.py", line 289, in lambda_handler
    return deploy_web_servers(event)
  File "/var/task/deploy_web_lambda.py", line 172, in deploy_web_servers
    latest_ami = find_latest_ami_name(region, "HVM64")
  File "/var/task/deploy_web_lambda.py", line 100, in find_latest_ami_name
    Owners=[owner]
…
botocore.exceptions.ClientError: An error occurred (UnauthorizedOperation) when calling the DescribeImages operation: You are not authorized to perform this operation.
```

Reply:
Well spotted. Indeed permission needed.